### ISSUE TITLE
Incorrect or correct docblock?

### ISSUE THREAD

**Issue by danherd** [laravel/ideas#1365]
I tried using [psalm](https://github.com/vimeo/psalm) to do some static analysis of a Lumen app and found it was throwing an error with my call to the `handle` method in the `Illuminate\Auth\Middleware\Authenticate` class (see [this issue here](https://github.com/vimeo/psalm/issues/1027)).

Here's the method in full:

```
    /**
     * Handle an incoming request.
     *
     * @param  \Illuminate\Http\Request  $request
     * @param  \Closure  $next
     * @param  string[]  ...$guards
     * @return mixed
     *
     * @throws \Illuminate\Auth\AuthenticationException
     */
    public function handle($request, Closure $next, ...$guards)
    {
        $this->authenticate($request, $guards);

        return $next($request);
    }
```

Now my call to the method is as follows:

`$this->authenticate->handle($request, $next, 'api');`

Which works, but psalm errors because it expects the third parameter to be an array. Technically, it will end up as an array because of the three dots indicating there could be multiple guards. However, they are passed as separate arguments - each one being a string. If I change the relevant docblock line to:

`     * @param  string  ...$guards`

psalm doesn't flag it up as an error.

So, in summary, should I submit a PR to Laravel to change any instance where a method is expecting variable arguments from having `[]` at the end? Or is psalm the one that is in the wrong here?

**Comment by 36864** [comment]
>So, in summary, should I submit a PR to Laravel to change any instance where a method is expecting variable arguments from having [] at the end? 

Blanket changes like that should be avoided. The best thing to do would be to verify what the method actually accepts (`string ...$guards` or `string[] ...$guards`) and correct it case-by-case. It might just be that the method accepts both versions, and the docblock might need to be corrected to `string ...$guards|string[] ...$guards`.

**Comment by danherd** [comment]
Thanks
